src/PaymentGateways/Stripe/ApplicationFee.php

```diff
@@ -28,14 +29,21 @@ class ApplicationFee {
 	 *
 	 * @unreleased
 	 *
+	 * @param string $accountId
+	 *
 	 * @return bool
 	 */
-	public static function canAddFee() {
+	public static function canAddFee( $accountId ) {
 		$gate = new static();
 
-		return ! ( $gate->hasLicense()
-			|| $gate->isStripeProAddonActive()
-			|| $gate->isStripeProAddonInstalled( get_plugins() ) );
+		/* @var $accountDetailRepository AccountDetail */
+		$accountDetailRepository = give( AccountDetail::class );
+		$stripeAccountDetail     = $accountDetailRepository->getAccountDetail( $accountId );
+
+		return $gate->isStripeProAddonActive()
```

Comment:
@ravinderk I think this reverses the logic of some of the gates from the previous commit - could you review this part?

Reply:
@kjohnson you are right. I will fix it.

Reply:
@kjohnson condition has been fixed